kubetest2-gke/deployer/common.go

```diff
@@ -74,3 +95,22 @@ func (d *deployer) initialize() error {
 
 	return nil
 }
+
+func filterMultiProjectClusterName(clusters []string, projectID int) ([]string, error) {
+	result := make([]string, 0)
+	for _, clusterName := range clusters {
+		parts := strings.Split(clusterName, "-")
```

Comment:
This is a breaking change. The separator should be something that the gke cluster name doesn't allow by default.

Reply:
This function will only be called `if numProjects > 1`, for single project, it will still be using the full name as the cluster name, so it's backward compatible.

Reply:
we should allow multi-cluster names to be any possible valid gke cluster name. 
otoh the choice of separator for the project suffix can be anything else.

Reply:
Makes sense.
Changed the separator to `:` which is not allowed in gke cluster names.